### PR TITLE
Fix process-node-related bugs

### PR DESCRIPTION
**Pull request by edan-bainglass**
Based on #946

This PR handles a couple of bugs related to the process node, specifically loading node views and managing the process state.

### CODE REVIEW

**Comment by edan-bainglass** [comment]
@superstar54 rebased. Please review once tests finish 🙏

File: src/aiidalab_qe/app/result/__init__.py [hunk 1]
```diff
@@ -189,6 +189,8 @@ def _update_node_view(self, nodes, refresh=False):
             self.node_view_container.children = [self.node_view_loading_message]
             self.node_view = self._create_workchain_viewer(node)
             self.node_views[node.uuid] = self.node_view
+        else:
```
Comment (superstar54) [inline src/aiidalab_qe/app/result/__init__.py:192]: Could you add a comment on what this case means and why return?

Comment (edan-bainglass) [reply src/aiidalab_qe/app/result/__init__.py:192]: This guard was intended to cover cases where you select a non-`QeAppWorkChain` `WorkChainNode`. We decided not to change the rendered view in this case. However, I think it is best instead to render a "No viewer available for this node" message until one is developed (if we decide to do so). This is now implemented.

Comment (superstar54) [reply src/aiidalab_qe/app/result/__init__.py:192]: Thanks! Could you add it as coment lines in the code?

Comment (edan-bainglass) [reply src/aiidalab_qe/app/result/__init__.py:192]: The current implementation is now self-documenting, no? If you're not registered in the cache AND ARE a `WorkChainNode` BUT ARE NOT the `QeAppWorkChain`, then you don't have a viewer. A comment would be redundant at this point.

Comment (superstar54) [reply src/aiidalab_qe/app/result/__init__.py:192]: I didn't notice you pushed a new commit to replace the `return` with a message. Now, it's clear in the code.

**Review by superstar54 — APPROVED** [review]
LGTM!

**Pull request merged**